front/src/common/BootstrapSNCF/InputGroupSNCF.tsx

```diff
@@ -70,36 +68,34 @@ export default function InputGroupSNCF({
   inputDataTestId,
 }: Props) {
   const [isDropdownShown, setIsDropdownShown] = useState(false);
-  const [selected, setSelected] = useState(
-    value
-      ? {
-          label: title,
-        }
-      : { id: options[0].id, label: options[0].label, unit: options[0].unit }
-  );
+  const [selectedUnit, setSelectedUnit] = useState<Option>({
+    id: options[0].id,
+    label: options[0].label,
+    unit: options[0].unit,
+  });
+
   const textAlignmentClass = textRight ? 'right-alignment' : 'left-alignment';
 
   useEffect(() => {
-    const selectedOption = options?.find((option) => option.id === type);
-
-    setSelected({
+    // Check if we can find the unit in the options id (allowances) or label (rolling stock editor)
```

Comment:
It would be better to only use the id, why is it not possible ?

Reply:
The string I pass in the unit prop comes from the rollingstockValues current parameter unit and is not "id formatted" to match the option id. It would require to add another property in the rollingstockValues state to handle the id and I didn't think it was worth just for that.